dnsmasq: fails to cross-compile

**Describe the bug**
`dnsmasq` fails to cross-compile on atleast `armv7l` and `aarch64`.

it fails with:
```
dbus.c:21:10: fatal error: dbus/dbus.h: No such file or directory
   21 | #include <dbus/dbus.h>
      |          ^~~~~~~~~~~~~
```

**To Reproduce**
Steps to reproduce the behavior:
- `nix-build https://github.com/NixOS/nixpkgs/archive/d8d19f249793966d1cce391d54203cc717764e63.tar.gz -A pkgsCross.armv7l-hf-multiplatform.dnsmasq`
- `nix-build https://github.com/NixOS/nixpkgs/archive/d8d19f249793966d1cce391d54203cc717764e63.tar.gz -A pkgsCross.aarch64-multiplatform.dnsmasq`

**Expected behavior**
dnsmasq is successfully cross-compiled.

**Notify maintainers**
@edolstra @fpletz @globin 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.7.2, NixOS, 20.09pre229870.22c98819ccd (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"nixos-20.09pre229870.22c98819ccd, nixos-20.03-20.03.2176.a84b797b28e"`
 - channels(betaboon): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

--- a/pkgs/tools/networking/dnsmasq/default.nix
+++ b/pkgs/tools/networking/dnsmasq/default.nix
@@ -1,5 +1,5 @@
 { lib, stdenv, fetchurl, pkg-config, dbus, nettle, fetchpatch
-, libidn, libnetfilter_conntrack }:
+, libidn, libnetfilter_conntrack, buildPackages }:
 
 with lib;
 let
@@ -33,6 +33,7 @@ stdenv.mkDerivation rec {
     "BINDIR=$(out)/bin"
     "MANDIR=$(out)/man"
     "LOCALEDIR=$(out)/share/locale"
+    "PKG_CONFIG=${buildPackages.pkg-config}/bin/${buildPackages.pkg-config.targetPrefix}pkg-config"
   ];
 
   hardeningEnable = [ "pie" ];

